app/searchForm/index.js

```diff
@@ -0,0 +1,34 @@
+class searchFormController {
+    /*@ngInject*/
+    constructor(gitSearchService) {
+        this.query = '';
+        this.gitSearchService = gitSearchService;
+    }
+
+    getRequestParams() {
+        let parsedParams = {};
+        let params = this.query.split('+');
+        params.forEach((parameter) => {
+            parameter = parameter.trim();
+            let subParams = parameter.split(':');
+            if (subParams.length > 1) {
+                let trimmedParams = subParams.map(Function.prototype.call, String.prototype.trim);
```

Comment:
I was just looking for a neat solution for trimming array elements without any values enumeration and I came across [this] (http://stackoverflow.com/questions/19293997/javascript-apply-trim-function-to-each-string-in-an-array).
It hurt my brain a bit while trying to understand how it works, but I believe that I got it ^_^